annotation/annotation/tasks/services.py

```diff
@@ -62,6 +62,33 @@
     ValidationSchema,
 )
 
+
+class LRU(OrderedDict):
+    def __init__(self, capacity):
+        self.capacity = capacity
+
+    def __contains__(self, item):
+        if super().__contains__(item):
+            self.move_to_end(item)
+            return True
+        else:
+            return False
+
+    def __getitem__(self, key):
+        if key in self:
+            self.move_to_end(key)
+            return super().__getitem__(key)
+        raise KeyError
+
+    def __setitem__(self, key, value):
+        if key in self:
+            self.move_to_end(key)
+        super().__setitem__(key, value)
+        if len(self) > self.capacity:
+            oldest = next(iter(self))
+            del self[oldest]
```

Comment:
```suggestion
            self.popitem(last=False)
```
because you move everything new to the right, meaning oldest is on the left

Reply:
I am getting a KeyError whenever I try to use the suggested implementation. It seems like that popitem calls the __getitem__ method after it has already removed the item from the dictionary. An error occurs in the __getitem__ method, after checking if the key is in the dictionary. 
```
    def test_lru():
        cache = services.LRU(2)
        cache[1] = "test1"
        cache[2] = "test2"
>       cache[3] = "test3"

tests\test_tasks_services.py:1481:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
annotation\tasks\services.py:84: in __setitem__
    super().popitem(last=False)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = LRU([(2, 'test2'), (3, 'test3')]), key = 1

    def __getitem__(self, key):
        if key in self:
>           self.move_to_end(key)
E           KeyError: 1

annotation\tasks\services.py:73: KeyError
```

Reply:
I guess it happens because we reimplement dunder methods instead of using normal names (like get, set). A lot of side effects